Implementation/Services/index.md

```diff
@@ -0,0 +1,510 @@
+```
+
+and insde a UrlProvider the GetUrl method has the current UmbracoContext injected:
+```csharp
+   public override UrlInfo GetUrl(UmbracoContext umbracoContext, IPublishedContent content, UrlProviderMode mode, string culture, Uri current)
+        {
+        var someContent = umbracoContext.ContentCache.GetById(1234);
+```
+
+## Custom Services and Helpers
+
+When implementing an Umbraco site, it is likely to have to execute similar code that accesses or operates on Umbraco data, in multiple places, perhaps using the Core Management Services or Umbraco Helpers.
+For example; Getting a list of the latest News Articles, or building a link to the site's News Section or Contact Us page. It's easy to repeat this kind of logic in multiple places, Views, Partial Views / Controllers etc, which is fine, but it's generally considered good practice to consolodate this logic into a single place.
+
+### Extension methods
+
+One option is to add 'Extension Methods' to the `UmbracoHelper` class.
+```csharp
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web;
+
+namespace Umbraco8.Extensions
+{
+    public static class UmbracoHelperExtensions
+    {
+            //NewsCategories //NewsSection
+            public static IPublishedContent GetNewsSection(this UmbracoHelper umbracoHelper)
+            {
+            return umbracoHelper.ContentSingleAtXPath("root/homePage/newsSection");
+            }
+     }
+}
+```
+
+anywhere there is reference to the UmbracoHelper, and a reference is added to the namespace the extension belongs to, it's possible to write `Umbraco.GetNewsSection()` - for a large site though, this can get a little bit out of control!
+
+### Custom Services and Helpers
+
+Another option, is to make use of the underlying DI framework, and create custom Services and Helpers, that in turn can have the 'core' Management Services and Umbraco Helpers injected into them - (the same rules about transient services apply!). This approach enables the grouping together of similar methods within a suitably named service, and promotes the possibility of testing this custom logic outside of controllers and views.
+
+In this example, we create a custom helper service, that's responsible for finding key pages within a site, eg the News Section or the Contact Us page - these methods will commonly be called in different places throughout the site, and it's great to encapsulate the logic to retrieve them in a single place - we'll call this helper service: SiteHelperService
+
+Create an interface to define the service:
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using Umbraco.Core.Models.PublishedContent;
+
+namespace Umbraco8.Services
+{
+        public interface ISiteHelperService
+    {
+        IPublishedContent GetNewsSection();
+        IPublishedContent GetContactUsPage();
+    }
+}
+
+Create the concrete service class that implements the interface:
```

Comment:
Forgot to close the code example above & thus you have a single text line in the code example, that people are bound to copy & paste